runtime/Cpp/runtime/src/atn/ATNDeserializer.cpp

```diff
@@ -422,7 +458,7 @@ ATN ATNDeserializer::deserialize(const std::vector<uint16_t>& input) {
 
           size_t ruleIndex = static_cast<ActionTransition *>(transition)->ruleIndex;
           size_t actionIndex = static_cast<ActionTransition *>(transition)->actionIndex;
-          Ref<LexerCustomAction> lexerAction = std::make_shared<LexerCustomAction>(ruleIndex, actionIndex);
+          std::shared_ptr<LexerCustomAction> lexerAction = std::make_shared<LexerCustomAction>(ruleIndex, actionIndex);
```

Comment:
From a readability point of view, `Ref` by itself is not very meaningful. It could be std::reference_wrapper, std::shared_ptr, or a custom rolled reference counted type. By using std::shared_ptr it is much more clear to the reader as almost all writers of C++ know what that is, nobody knows what `Ref` is.

Reply:
Ah, I double-checked. Its `__` which is reserved and `_[A-Z]...`, not single underscores by themselves. Though Microsoft's C library uses single underscores a lot. Switching back.

Reply:
Sounds a bit odd, given that `std::shared_ptr` is much worse to read. The `Ref` macro has been introduced exactly for the reason of simplifying expressions. Honestly, I see no reason to accept that change.

Reply:
Switched back to Ref since it is used everywhere, but I'll leave it with this. If you are reading a source file at see `Ref` nobody knows what that is until they go find the definition. However most C++ programmers know what `std::shared_ptr` is.